sql/core/src/test/scala/org/apache/spark/sql/internal/CatalogSuite.scala

```diff
@@ -743,4 +743,35 @@ class CatalogSuite extends SharedSparkSession with AnalysisTest with BeforeAndAf
     val catalogName2 = "catalog_not_exists"
     assert(!spark.catalog.databaseExists(Array(catalogName2, dbName).mkString(".")))
   }
+
+  test("three layer namespace compatibility - current database") {
```

Comment:
hmm I think you can also update `currentDatabase` API and the test will be simplified:

`setCurrentNamespace` then check `currentNamespace`

https://github.com/databricks/runtime/blob/ca9cfac39f6ca160091cc3a1d7727ee0f784d044/sql/core/src/main/scala/org/apache/spark/sql/internal/CatalogImpl.scala#L67

Reply:
and catalog manager has a `currentNamespace`: https://github.com/apache/spark/blob/a3fdf3b107b62089890a4246c168ca516dff5d44/sql/catalyst/src/main/scala/org/apache/spark/sql/connector/catalog/CatalogManager.scala#L92.

So for changing `currentDatabase`, it could be as easy as calling the catalog manager `currentNamespace`